src/ddprof_module_lib.cc

```diff
@@ -15,12 +15,33 @@
 #include <fcntl.h>
 #include <gelf.h>
 #include <libelf.h>
-#include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 
 namespace ddprof {
 
+class FileDescriptorHolder {
```

Comment:
This could be a use case for [unique_resource](https://en.cppreference.com/w/cpp/experimental/unique_resource), but I don't know if it is widely available...